libraries/c_sdk/standard/ble/src/iot_ble_gap.c

```diff
@@ -876,9 +900,63 @@ BTStatus_t IotBle_Init( void )
         }
     }
 
+    /*
+     * If a BLE device name is provided in the config, use it to set the local BLE device name.
+     */
+
+    if( ( pDeviceName != NULL ) && ( strcmp( pDeviceName, "" ) != 0 ) )
+    {
+        status = IotBle_SetDeviceName( pDeviceName, strlen( pDeviceName ) );
+    }
+
+    return status;
+}
+
+BTStatus_t IotBle_SetDeviceName( const char * pName,
+                                 size_t length )
+{
+    BTStatus_t status;
+
+    if( length <= IOT_BLE_MAX_DEVICE_NAME_LENGTH )
+    {
+        /* Copy the device name to a buffer, so that device name does not change between
+         * stack disable-enable cycles.
+         */
+        strncpy( bleDeviceName, pName, length );
+        bleDeviceName[ length + 1 ] = '\0';
+        status = eBTStatusSuccess;
+    }
+    else
+    {
+        status = eBTStatusNoMem;
+    }
+
+    if( status == eBTStatusSuccess )
+    {
+        if( isBLEOn )
```

Comment:
In the header iot_ble.h:649 it states:

> API can be used to set the device name before turning on BLE by calling IotBle_On(), or when BLE is running. When BLE is advertising mode...

Does that mean the name can be adjusted while BLE is off? If so, I'm confused how this call could modify the name while BLE is off, given this call won't modify the name unless `isBLEOn == true`

Reply:
API copies the device name to a buffer, so when BLE is turned on ( IotBle_On() ) it sets the stack property with the updated value.

Reply:
Ah, I see that now in IotBle_On. Got it